yt/data_objects/particle_filters.py

```diff
@@ -52,13 +52,15 @@ def apply(self, dobj):
             if f[0] != self.filtered_type: continue
             if tr.shape != filter.shape and tr.shape[0] != filter.shape[0]:
                 raise YTIllDefinedFilter(self, tr.shape, filter.shape)
-            elif filter.size == 0:
-                # Filtering empty set.  This keeps our dimensions correct.
-                # Otherwise we end up with out-of-axis and shape problems.
-                d = tr.copy() 
-            elif len(tr.shape) > len(filter.shape):
-                # Filter must always be 1D
-                d = tr[filter,:]
+            # I do not think these conditions will ever be true,
+            # am I good to remove them?
```

Comment:
I tried to come up with cases where the lines you deleted get hit and I'm pretty sure you're right and these branches can be deleted. For the first branch, I don't think it's possible to create an empty filter, because there needs to be a non-zero number of particles to have a particle field present in the dataset. Additionally, it's not possible to create a particle filter for a particle type that's not present in the dataset.

For the second branch, the check on line 53 means that that check in like 59 can't evaluate to True.

So yeah, just delete the commented out lines, along with your commented question here.

Reply:
done